XpringKit/PayID/PayIDClient.swift

```diff
@@ -71,7 +71,8 @@ public class PayIDClient {
       if
         let response = response,
         let paymentInformation = response.body {
-        completion(.success(paymentInformation.addressDetails))
+        // TODO(amiecorso): what if addresses is empty or has more than one address?
+        completion(.success(paymentInformation.addresses[0].addressDetails))
```

Comment:
Great point. The whole reason for the breaking change was to allow requesting `Accept: application/payid+json`, which returns the array of all addresses that PayID has. Does the SDK not support that right now?

Reply:
ah gotcha.  No the SDK does not support this right now in the sense that it is not set up to return an array of addresses.  If a `PayIDClient` was instantiated with the network `"payid"` as the argument to the constructor, then it would assemble and submit the content negotiation accept header.  I think we need to decide how to handle this header:

Reply:
It is tough for me to imagine why we'd expose something that let you get multiple addresses in the SDK. The binding here is 1:1 between PayIDClient and Network. (It's tough for me to visualize in general why anyone would want content negotiation, but that's out of band). You'd never ask a PayIDClient connected to "xrpl-test" for a "eth" address. Simply put, this method should never return more than 1 address and it is an error condition if the network request does. 

The way I'd do PayID negotiation on the client would probably be composing an array of PayIDClients into a `MultiPayIDClient` (Name TBD). You can also come up with a nicer API where you give us an array of strings and we make you the PayIDClients. They, you just need to iterate through each `PayIDClient` in `MultiPayIDClient` and return the first one that gives you an address. That certainly seems easier to me than crafting some bespoke headers and trying to interpret some non-standardized string array sent back from the server. This isn't hard to do but no one has asked for it yet and I'd want to hear a use case before we expand the PayID footprint further.  

The only case I can imagine is where you'd want to retrieve all addresses for a user. I'm not really sure where that lives... It doesn't make sense in the context of a `PayIDClient` and it doesn't make sense in a context of a `MultiPayIDClient`. That's probably a third class? Or a utility method? TBD, but it also wouldn't live in this 1:1 hierarchy. 

----

I'm supportive of PayID adding content-negotiation if that's what the team, or the powers that be, want. There are way easier ways to do this client side than content negotiation.

Reply:
> It is tough for me to imagine why we'd expose something that let you get multiple addresses in the SDK.

I guess the question is whether or not we want the SDK to be the "normal" way to interact with PayID. I don't know this for certain, but I suspect that most partners are rolling their own HTTP APIs rather than using the SDK right now.

I have two (theoretical) concrete scenarios that illustrate content negotiation and "get all addresses".

#### Content Negotiation

I have my wallet with BTC/ETH/XRP. We go out for dinner together, and you cover the bill to get the credit card points, and I want to pay you back. Neither of us care whether I pay you in BTC, ETH, or XRP, and I may not even know which currencies your wallet actually supports.

The UI for this wallet might even just be a "PAY" button with a input for a PayID.

I want to pay the least amount in fees when transferring money to you, so I make a payment with `Accept: xrpl, eth;q=0.9, btc;q=0.8`. I only need to make a single HTTP request instead of multiple, and I'm guaranteed to get the payment address that will cost me the least in fees (assuming XRPL < ETH < BTC for fees).

#### Get All Addresses

My wallet has a UI allowing me to select which currencies I can pay you in. However, it also wants to disable currencies that the beneficiary doesn't support. I make my `payid+json` request to you, find out you don't support XRPL, and disable the XRP selection on the UI.